internal/pipeline/executor.go

```diff
@@ -281,12 +281,13 @@ func adjustRelPath(resources []*yaml.RNode, relPath string) ([]*yaml.RNode, erro
 		return resources, nil
 	}
 	for _, r := range resources {
-		meta, err := r.GetMeta()
+		currPath, _, err := kioutil.GetFileAnnotations(r)
 		if err != nil {
 			return resources, err
 		}
-		if !strings.HasPrefix(meta.Annotations[kioutil.PathAnnotation], relPath) {
-			newPath := path.Join(relPath, meta.Annotations[kioutil.PathAnnotation])
+		// if currPath is relative to root pkg i.e. already has relPath, skip it
+		if !strings.HasPrefix(currPath, relPath+"/") {
```

Comment:
Shouldn't the path annotation be always relative to root package ? Why do we need to adjust that ?

Reply:
No, during hydration, we start from inward pkg first moving outward. So when we read (and hydrate) the resources, we read (and hydrate) them at it's pkg level (so the path is relative to its pkg) so we adjust it to the rootpkg because when we write back, we write it at the rootlevel.

So this function can get called multiple times as we move from inward towards outward. So first call to this function from the most inward pkg, will set the annotation relative to root and subsequent calls will always find the relative path there.

In our test case, `/` forward slash was missing that caused the issue.